src/Models/Address.php

```diff
@@ -107,6 +108,8 @@ protected static function booted(): void
         });
 
         static::saved(function (Address $address) {
+            Cache::forget('morph_to:' . $address->getMorphClass() . ':' . $address->id . ':name');
```

Comment:
In theory all morph_to:address:id cache keys have to be forgotten, not only :name. Also :name must only be forgotten if it got changed.